extension/fts/src/function/query_fts_index.cpp

```diff
@@ -28,32 +29,65 @@ namespace fts_extension {
 using namespace function;
 
 struct QueryFTSBindData final : public function::GDSBindData {
-    std::vector<std::string> terms;
-    uint64_t numDocs;
-    double_t avgDocLen;
+    std::shared_ptr<Expression> query;
     QueryFTSConfig config;
+    const FTSIndexCatalogEntry& entry;
     common::table_id_t outputTableID;
 
-    QueryFTSBindData(std::vector<std::string> terms, graph::GraphEntry graphEntry,
-        std::shared_ptr<binder::Expression> docs, uint64_t numDocs, double_t avgDocLen,
+    QueryFTSBindData(std::shared_ptr<Expression> query, graph::GraphEntry graphEntry,
+        std::shared_ptr<binder::Expression> docs, const FTSIndexCatalogEntry& entry,
         QueryFTSConfig config)
-        : GDSBindData{std::move(graphEntry), std::move(docs)}, terms{std::move(terms)},
-          numDocs{numDocs}, avgDocLen{avgDocLen}, config{config},
+        : GDSBindData{std::move(graphEntry), std::move(docs)}, query{std::move(query)},
+          config{config}, entry{entry},
           outputTableID{nodeOutput->constCast<NodeExpression>().getSingleEntry()->getTableID()} {}
     QueryFTSBindData(const QueryFTSBindData& other)
-        : GDSBindData{other}, terms{other.terms}, numDocs{other.numDocs},
-          avgDocLen{other.avgDocLen}, config{other.config}, outputTableID{other.outputTableID} {}
+        : GDSBindData{other}, query{other.query}, config{other.config}, entry{other.entry},
+          outputTableID{other.outputTableID} {}
 
     bool hasNodeInput() const override { return false; }
 
-    uint64_t getNumUniqueTerms() const;
+    std::vector<std::string> getTerms() const;
 
     std::unique_ptr<GDSBindData> copy() const override {
         return std::make_unique<QueryFTSBindData>(*this);
     }
 };
 
-uint64_t QueryFTSBindData::getNumUniqueTerms() const {
+std::vector<std::string> QueryFTSBindData::getTerms() const {
+    auto value = Value::createDefaultValue(query->dataType);
+    switch (query->expressionType) {
```

Comment:
Shall we abstract a function in ExpressionUtils that getValue for a given expression when it is LITERAL or PARAMETER? If so we might also refactor the previous one we did for vector index.